src/ol/format/esrijsonformat.js

```diff
@@ -0,0 +1,687 @@
+ * @private
+ * @return {EsriJSONPolyline} EsriJSON geometry.
+ */
+ol.format.EsriJSON.writeMultiLineStringGeometry_ =
+    function(geometry, opt_options) {
+  goog.asserts.assertInstanceof(geometry, ol.geom.MultiLineString,
+      'geometry should be an ol.geom.MultiLineString');
+  var hasZM = ol.format.EsriJSON.getHasZM_(geometry);
+  return /** @type {EsriJSONPolyline} */ ({
+    'hasZ': hasZM.hasZ,
+    'hasM': hasZM.hasM,
+    'paths': geometry.getCoordinates()
+  });
+};
+
+
+/**
+ * @param {ol.geom.Geometry} geometry Geometry.
+ * @param {olx.format.WriteOptions=} opt_options Write options.
+ * @private
+ * @return {EsriJSONMultipoint} EsriJSON geometry.
+ */
+ol.format.EsriJSON.writeMultiPointGeometry_ = function(geometry, opt_options) {
+  goog.asserts.assertInstanceof(geometry, ol.geom.MultiPoint,
+      'geometry should be an ol.geom.MultiPoint');
+  var hasZM = ol.format.EsriJSON.getHasZM_(geometry);
+  return /** @type {EsriJSONMultipoint} */ ({
+    'hasZ': hasZM.hasZ,
+    'hasM': hasZM.hasM,
+    'points': geometry.getCoordinates()
+  });
+};
+
+
+/**
+ * @param {ol.geom.Geometry} geometry Geometry.
+ * @param {olx.format.WriteOptions=} opt_options Write options.
+ * @private
+ * @return {EsriJSONPolygon} EsriJSON geometry.
+ */
+ol.format.EsriJSON.writeMultiPolygonGeometry_ = function(geometry,
+    opt_options) {
+  goog.asserts.assertInstanceof(geometry, ol.geom.MultiPolygon,
+      'geometry should be an ol.geom.MultiPolygon');
+  var hasZM = ol.format.EsriJSON.getHasZM_(geometry);
+  var coordinates = geometry.getCoordinates(false);
+  var output = [];
+  for (var i = 0; i < coordinates.length; i++) {
+    for (var x = coordinates[i].length - 1; x >= 0; x--) {
+      output.push(coordinates[i][x]);
+    }
+  }
+  return /** @type {EsriJSONPolygon} */ ({
+    'hasZ': hasZM.hasZ,
+    'hasM': hasZM.hasM,
+    'rings': output
+  });
+};
+
+
+/**
+ * @const
+ * @private
+ * @type {Object.<string, function(EsriJSONObject): ol.geom.Geometry>}
+ */
+ol.format.EsriJSON.GEOMETRY_READERS_ = {
+  'Point': ol.format.EsriJSON.readPointGeometry_,
+  'LineString': ol.format.EsriJSON.readLineStringGeometry_,
+  'Polygon': ol.format.EsriJSON.readPolygonGeometry_,
+  'MultiPoint': ol.format.EsriJSON.readMultiPointGeometry_,
+  'MultiLineString': ol.format.EsriJSON.readMultiLineStringGeometry_,
+  'MultiPolygon': ol.format.EsriJSON.readMultiPolygonGeometry_
+};
+
+
+/**
+ * @const
+ * @private
+ * @type {Object.<string, function(ol.geom.Geometry, olx.format.WriteOptions=): (EsriJSONGeometry)>}
+ */
+ol.format.EsriJSON.GEOMETRY_WRITERS_ = {
+  'Point': ol.format.EsriJSON.writePointGeometry_,
+  'LineString': ol.format.EsriJSON.writeLineStringGeometry_,
+  'Polygon': ol.format.EsriJSON.writePolygonGeometry_,
+  'MultiPoint': ol.format.EsriJSON.writeMultiPointGeometry_,
+  'MultiLineString': ol.format.EsriJSON.writeMultiLineStringGeometry_,
+  'MultiPolygon': ol.format.EsriJSON.writeMultiPolygonGeometry_
```

Comment:
Same comment above about quoted keys.  These would only need to be quoted if values were coming from some input you were reading.